fixes #325 by updating references to The Notary Project (#327)
notaryproject/notaryproject.dev: content/en/docs/overview.md

---
title: "Project overview"

> description: "An overview of The Notary Project"

Reviewer: Should we use capital `T`?

Author: Fixed

type: docs
weight: 1
---

{{% alert title="Important" color="info" %}}

> This article as well as the rest of this documentation describes the latest status of The Notary Project and refers to it as The Notary Project. For documentation of older versions and implementations, please refer to the [previous documentation](https://github.com/notaryproject/notary/tree/master/docs).

Reviewer: Should we use capital `T`?

Author: Fixed

> This article as well as the rest of this documentation describes the latest status of the Notary Project and refers to it as the Notary Project. For documentation of older versions and implementations, please refer to the [previous documentation](https://github.com/notaryproject/notary/tree/master/docs).

Reviewer 2: This article as well as the rest of this documentation describes the latest status of the Notary Project. For documentation of Notary TUF-based implementations, please refer to the [previous documentation](https://github.com/notaryproject/notary/tree/master/docs).

> This article as well as the rest of this documentation describes the latest status of the Notary Project and refers to it as the Notary Project. For documentation of older versions and implementations, please refer to the [previous documentation](https://github.com/notaryproject/notary/tree/master/docs).

Reviewer 2: Just correct a typo on @yizha1 's comment above.

> This article as well as the rest of this documentation describes the latest status of the Notary Project and refers to it as the Notary Project. For documentation of older versions and implementations, please refer to the [previous documentation](https://github.com/notaryproject/notary/tree/master/docs).

Reviewer: Thanks @FeynmanZhou , we should use lowercase and code `notary` when referring to TUF-based implementation. So it should be `notary` TUF-based implementation.

> This article as well as the rest of this documentation describes the latest status of the Notary Project and refers to it as the Notary Project. For documentation of older versions and implementations, please refer to the [previous documentation](https://github.com/notaryproject/notary/tree/master/docs).

Author: Addressed in separate change

{{% /alert %}}

## Project Status

> The Notary Project is in early development and design documents should not be considered final.

Reviewer: I can understand T is capital where it is at the beginning of the sentence.

Author: Leaving as is

> The Notary Project is in early development and design documents should not be considered final.

Author: Incorporated some of this feedback.

Please refer to the [milestones](https://github.com/notaryproject/notaryproject/milestones) or [attend]({{< ref "/community" >}}) the weekly meetings for details on the roadmap.

> ## The Notary Project Overview

Reviewer: ```suggestion
## Notary Project Overview
```

Author: fixed

> ![The Notary Project scenarios](/docs/notary-e2e-scenarios.svg)

Reviewer: ```suggestion
![Notary Project scenarios](/docs/notary-e2e-scenarios.svg)
```

Author: fixed

> The Notary Project provides for multiple signatures of an OCI Artifact (including container images) to be persisted in an [OCI distribution v1.1 compliant][oci-distribution-conformance] registry.

Author: This feedback came from @yizha1: https://github.com/notaryproject/notaryproject.dev/pull/327#discussion_r1275757984

> The Notary Project provides for multiple signatures of an OCI Artifact (including container images) to be persisted in an [OCI distribution v1.1 compliant][oci-distribution-conformance] registry.

Reviewer: @zr-msft is correct, we can remove the version. Then it is `OCI distribution compliant`

> The Notary Project provides for multiple signatures of an [OCI Artifact][oci-artifacts] (including container images) to be persisted in an [OCI conformant][oci-distribution-conformance] registry.

Reviewer: ```suggestion
Notary Project provides for multiple signatures of an OCI artifact (including container images) to be persisted in an [OCI distribution v1.1 compliant][oci-distribution-conformance] registry.
```

Author: fixed

Artifacts are signed (`notation sign`) with private keys, and validated with public keys (`notation verify`).
To support user deployment flows, signing an OCI Artifact will not change the `@digest` or `artifact:tag` reference.
To support content movement across multiple certification boundaries, artifacts and their signatures will be easily copied within and across [OCI conformant][oci-distribution-conformance] registries.

> To deliver on the Notary Project goals of cross registry movement of artifacts with their signatures, changes to several projects are anticipated, including [OCI distribution-spec][oci-distribution-spec], [CNCF Distribution][cncf-distribution], [OCI Artifacts][oci-artifacts], [ORAS][oras] with further consumption from projects including [containerd][containerd], [OPA][opa], [Gatekeeper][gatekeeper] and the [docker client][docker-client].

Reviewer: ```suggestion
To deliver on the Notary Project goals of cross registry movement of artifacts with their signatures, changes to several projects are anticipated, including [OCI distribution-spec][oci-distribution-spec], [CNCF Distribution][cncf-distribution], [ORAS][oras] with further consumption from projects including [containerd][containerd], [OPA][opa], [Gatekeeper][gatekeeper] and the [docker client][docker-client].
```

Reviewer: OCI artifact has been archived, so please also remove the reference at the bottom of this file

```
[oci-artifacts]:                https://github.com/opencontainers/artifacts
[oci-artifact-manifest]:        https://github.com/opencontainers/artifacts/pull/29
```

Author: fixed

> To deliver on The Notary Project goals of cross registry movement of artifacts with their signatures, changes to several projects are anticipated, including [OCI distribution-spec][oci-distribution-spec], [CNCF Distribution][cncf-distribution], [OCI Artifacts][oci-artifacts], [ORAS][oras] with further consumption from projects including [containerd][containerd], [OPA][opa], [Gatekeeper][gatekeeper] and the [docker client][docker-client].

Reviewer: Should we use capital T?

Author: fixed

- Copy a container image to a private registry, verifying the source then adding a verification signature
- run one or more verification processes, then sign with the ACME Rockets key

## The Notary Project Journey

> Notary [kicked off in December of 2019][notaryv2-kickoff] with a [broad range of attendees][kickoff-attendees].

Reviewer: ```suggestion
Notary Project [kicked off in December of 2019][notary-project-kickoff] with a [broad range of attendees][kickoff-attendees].
```

Author: fixed

## Top Areas of Focus

To complete The Notary Project, three key areas of focus were identified.

> ### Definition of a Notary Signature

Reviewer: Notary Project signature

Author: fixed

#### Registry Discovery

Registry discovery of linked artifacts enables finding a signature, based on the target artifact.
In the Notary Project example, the ACME Rockets production servers must be capable of efficiently finding the ACME Rockets signature for the `net-monitor:v1` image.
Once the signature is identified, through a content addressable digest, the Notary Project client, such as the Notation CLI, may validate the signature.

### Key and Certificate Management

Key Management involves the following key scenarios:

- Signing with private keys
- Publishing and discovery of public keys for consumers to validate signatures
- Key revocation, including support for air-gapped environments

Private key management is beyond the scope of the Notary Project effort, as companies have well defined practices that are internal to their software development.

> Publishing and discovery of public keys should be easy for consumers to acquire, however, the Notary Project will not implicitly support a **T**rust **o**n **F**irst **U**se (TOFU) model.

Reviewer 3: ```suggestion
Publishing and discovery of public keys should be easy for consumers to acquire, however, the Notary Project specifications will not support a **T**rust **o**n **F**irst **U**se (TOFU) model.
```

Author: removed overview.md from PR per @yizha1

#### Registry Discovery

Registry discovery of linked artifacts enables finding a signature, based on the target artifact.
In the Notary Project example, the ACME Rockets production servers must be capable of efficiently finding the ACME Rockets signature for the `net-monitor:v1` image.

> Once the signature is identified, through a content addressable digest, the Notary Project client may validate the signature.

Reviewer 3: ```suggestion
Once the signature is identified, through a content addressable digest, the Notary Project clients (e.g. Notation) may validate the signature.
```

> ### Key Management

Reviewer 3: ```suggestion
### Key and Certificate Management
```

Author: fixed

- Publishing and discovery of public keys for consumers to validate signatures
- Key revocation, including support for air-gapped environments

> Private key management is beyond the scope of The Notary Project effort, as companies have well defined practices that are internal to their software development.

Reviewer: ```suggestion
Private key management is beyond the scope of the Notary Project effort, as companies have well defined practices that are internal to their software development.
```

Author: fixed

> Publishing and discovery of public keys should be easy for consumers to acquire, however, the Notary Project will not implicitly support a **T**rust **o**n **F**irst **U**se (TOFU) model.

Reviewer 3: ```suggestion
Publishing and discovery of public keys should be easy for consumers to acquire, however unlike the prior notary implementation, the new Notary Project specifications will not implicitly support a **T**rust **o**n **F**irst **U**se (TOFU) model.
```

Author: I think the text should remain as is.

We followed the design patterns of other large, complex projects like Antoni Gaudí's design of [The Sagrada Familia](https://simple.wikipedia.org/wiki/Sagrada_Fam%C3%ADlia).
The [sketch, prototype, build approach](https://stevelasker.blog/2020/07/31/sketch-prototype-build/) would enable the various experts to focus on their component, while understanding where they plug-into other components of the design.

As a result, we identified the following stages of the Notary Project effort:

1. Define Requirements

> 1. Build Prototypes

Reviewer 3: ```suggestion
1. Build Prototypes using Notation
```

Author: fixed

1. Validate prototypes - learning, refining requirements, iterating prototypes

> 1. Author a Notary Spec

Reviewer 3: ```suggestion
1. Author new Notary Project specifications
```

Author: nice catch, fixed.

> See [the Notary Project on GitHub](https://github.com/notaryproject/notaryproject/tree/main/status-updates) for updates to the stages of development and areas of focus. 

Reviewer 3: suggestion. either we update content on this link, or we point users back to our Github milestones/roadmap.

Reviewer: ```suggestion
See milestones of respective Notary Project sub-projects for updates to development and areas of focus. 
```

Author: fixed
